src/v/cloud_roles/refresh_credentials.cc

```diff
@@ -235,6 +236,42 @@ http::client cloud_roles::refresh_credentials::impl::make_api_client() const {
       _as};
 }
 
+cloud_roles::refresh_credentials cloud_roles::make_refresh_credentials(
+  model::cloud_credentials_source cloud_credentials_source,
+  ss::gate& gate,
+  ss::abort_source& as,
+  credentials_update_cb_t creds_update_cb,
+  s3::aws_region_name region,
+  std::optional<endpoint> endpoint,
+  retry_params retry_params) {
+    switch (cloud_credentials_source) {
+    case model::cloud_credentials_source::config_file:
+        vlog(
+          clrl_log.error,
+          "invalid request to create refresh_credentials for static "
+          "credentials");
+        throw std::invalid_argument(fmt_with_ctx(
+          fmt::format, "cannot generate refresh with static credentials"));
+    case model::cloud_credentials_source::aws_instance_metadata:
+        return make_refresh_credentials<aws_refresh_impl>(
+          gate,
+          as,
+          std::move(creds_update_cb),
+          std::move(region),
+          std::move(endpoint),
+          retry_params);
+    default:
```

Comment:
since `case model::cloud_credentials_source::aws_instance_metadata` is a enum class, you don't need to use the default case because the compiler will enforce that you cover all the cases.

Reply:
fixed